src/osmet/file.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright 2020 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fs::{File, OpenOptions};
+use std::io::{self, BufReader, BufWriter, Read};
+use std::path::Path;
+
+use error_chain::bail;
+use serde::{Deserialize, Serialize};
+use tempfile::NamedTempFile;
+use xz2::read::XzDecoder;
+
+use super::*;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub(super) struct OsmetFile {
+    magic: [u8; 8],
+    major: u16,
+    minor: u16,
```

Comment:
Actually, the data structures don't have any extension points, so feature flags don't make sense either.  Maybe just a single version number?

But also, if I add a field to `struct Osmet` and try to read an existing osmet file, I get:

```
Error: failed to deserialize osmet file
Caused by: io error: failed to fill whole buffer
```

because bincode [doesn't handle changes to struct definitions](http://tyoverby.com/posts/bincode_release.html#should-i-use-bincode).  We could have a separate, fixed header that includes just the magic number and version information (more limited than the one contemplated by https://github.com/coreos/coreos-installer/pull/187#discussion_r402690732).  Or we could use something other than bincode.

Reply:
Yup, separating magic number + version info into a fixed header makes sense to me.

As mentioned in https://github.com/coreos/coreos-installer/pull/187#issuecomment-607445367, backwards compatibility isn't a concern here. IOW, maybe it's not even worth a "version" field per se.

How about just a hash of the coreos-installer [version field](https://github.com/coreos/coreos-installer/blob/893f87820d09eca0a92ecf04c7f02536f930d76f/Cargo.toml#L9) itself? The main thing I want to enforce is making sure that the same code that did the packing is also doing the unpacking and error out otherwise.

Reply:
Could do, though that won't catch semantic changes among development builds within a release.  I'd be inclined to stick to a single version field (defined in a symbolic constant) and hand-bump it on changes.